src/ConfigFile.c

```diff
@@ -549,29 +525,54 @@ void Ros_ConfigFile_ValidateCriticalSettings()
 
     //-----------------------------------------
     //Verify agent is on my subnet (or a gateway is specified)
+    //Save the LAN port being used for ROS traffic
     BOOL bAgentOnMySubnet = FALSE;
 
     //check first lan port
     STATUS status = Ros_ConfigFile_HostOnNetworkInterface(
         g_nodeConfigSettings.agent_ip_address, ROS_USER_LAN1, &bAgentOnMySubnet);
-    motoRosAssert_withMsg(status == OK, SUBCODE_CONFIGURATION_AGENT_ON_NET_CHECK,
+    motoRosAssert_withMsg(status == OK, SUBCODE_CONFIGURATION_AGENT_ON_NET_CHECK0,
         "Host on NIC check 1");
+    //This will be re-assigned later if it is not on the right subnet
+    g_Ros_Controller.rosTrafficLanPort = ROS_USER_LAN1;
 
 #if defined (YRC1000)
     if (!bAgentOnMySubnet)
     {
         //check second lan port
         status = Ros_ConfigFile_HostOnNetworkInterface(
             g_nodeConfigSettings.agent_ip_address, ROS_USER_LAN2, &bAgentOnMySubnet);
-        motoRosAssert_withMsg(status == OK, SUBCODE_CONFIGURATION_AGENT_ON_NET_CHECK,
+        motoRosAssert_withMsg(status == OK, SUBCODE_CONFIGURATION_AGENT_ON_NET_CHECK1,
             "Host on NIC check 2");
+        //If not on the subnet, it will alarm and cease before this matters...
+        g_Ros_Controller.rosTrafficLanPort = ROS_USER_LAN2;
     }
 #endif
 
     motoRosAssert_withMsg(bAgentOnMySubnet,
         SUBCODE_CONFIGURATION_INVALID_AGENT_SUBNET,
         "Agent IP on wrong subnet");
 
+    //Save the MAC address of the port being used for ROS traffic
+    status = Ros_GetMacAddress(g_Ros_Controller.rosTrafficLanPort, g_Ros_Controller.rosTrafficMacAddr);
+    if (status != OK)
+    {
+        Ros_Debug_BroadcastMsg("%s: Ros_GetMacAddress: iface: %d; error: %d",
+            __func__, g_Ros_Controller.rosTrafficLanPort, status);
+        motoRosAssert_withMsg(FALSE, SUBCODE_CONFIGURATION_FAIL_MP_NICDATA,
+            "Must enable ETHERNET function");
+    }
+
+    if (strlen(g_nodeConfigSettings.node_name) == 0)
```

Comment:
Since you have a defined constant for `PLACEHOLDER_NODE_NAME`, it's probably better to explicitly compare to that.